echoregions/regions2d/regions2d.py

```diff
@@ -501,3 +501,226 @@ def mask(
+        break_str = transect_dict["break"]
+        resume_str = transect_dict["resume"]
+        end_str = transect_dict["end"]
+        transect_strs = [start_str, break_str, resume_str, end_str]
+
+        # Check that there are 4 unique transect strings
+        if len(transect_strs) != len(set(transect_strs)):
+            raise ValueError(
+                "There exist duplicate values in transect_dict. \
+                             All values must be unique"
+            )
+
+        # Check that transect strings are of same length
+        for transect_str in transect_strs:
+            if len(transect_strs[0]) != len(transect_str):
+                raise ValueError("Transect strings must be of same length.")
+
+        # Create transect_df
+        region_df = self.data.copy()
+        transect_df = region_df.loc[
+            region_df.loc[:, "region_name"].str.startswith(start_str)
+            | region_df.loc[:, "region_name"].str.startswith(break_str)
+            | region_df.loc[:, "region_name"].str.startswith(resume_str)
+            | region_df.loc[:, "region_name"].str.startswith(end_str)
+        ].copy()
+
+        # Create a new column which stores the transect_type without the transect number
+        transect_df.loc[:, "transect_type"] = transect_df.loc[:, "region_name"].str[
+            : len(transect_str)
+        ]
+
+        # Checking the maximum width of a transect log region bbox.
+        # If over an hour, throw an error.
+        max_time = (
+            transect_df["region_bbox_right"] - transect_df["region_bbox_left"]
+        ).max()
+        max_time_hours = max_time.total_seconds() / (60 * 60)
+        if max_time_hours > 1.0:
+            raise ValueError(
+                "Maximum width in time of transect log region bboxs is \
+                             too large i.e. over an hour."
+            )
+
+        # Drop time duplicates
+        transect_df = transect_df.drop_duplicates(subset=["region_bbox_left"])
+
+        # Sort the dataframe by datetime
+        transect_df = transect_df.sort_values(by="region_bbox_left")
+
+        # Create new shifted columns with the next transect log type and next region
+        # bbox left datetime value.
+        transect_df.loc[:, "transect_type_next"] = transect_df.loc[
+            :, "transect_type"
+        ].shift(-1)
+        transect_df.loc[:, "region_bbox_left_next"] = transect_df.loc[
+            :, "region_bbox_left"
+        ].shift(-1)
+
+        # Check if for all transects, there exists 1 start_str transect type.
+        if not (
+            transect_df.groupby("file_name").apply(
+                lambda x: x[x["transect_type"] == start_str].count()
+            )["file_name"]
+            == 1
+        ).all():
+            raise ValueError(
+                f"Each transect must contain a single {start_str} transect_type. \
+                             There exists transect that does not contain a single {start_str}."
```

Comment:
it might be ok to not have ST/ET, and then assume that the beginning or end of the Sv file correspond to them but show a warning. I am not sure how many of those special cases we will encounter across the dataset.

Reply:
Yeah, this sounds like a better idea. I'll make a count of this case when I run the data in 2013 and later masking.